app/src/main/java/org/zalando/nakadi/config/WebConfig.java

```diff
@@ -83,35 +83,40 @@ public FilterRegistrationBean traceRequestFilter() {
     }
 
     @Bean
-    public FilterRegistrationBean monitoringRequestFilter() {
+    public FilterRegistrationBean flowIdRequestFilter() {
         return createFilterRegistrationBean(
-                new MonitoringRequestFilter(metricRegistry, perPathMetricRegistry, authorizationService),
+                new FlowIdRequestFilter(),
                 Ordered.HIGHEST_PRECEDENCE + 10);
     }
 
+    //
+    // Plugins may add more filters in the middle.
+    //
+    // In particular, the client id from the token is set by a plugin: therefore, put any filters
+    // that use AuthorizationService under this line.
+    //
+
     @Bean
-    public FilterRegistrationBean flowIdRequestFilter() {
+    public FilterRegistrationBean monitoringRequestFilter() {
         return createFilterRegistrationBean(
-                new FlowIdRequestFilter(),
-                Ordered.HIGHEST_PRECEDENCE + 20);
+                new MonitoringRequestFilter(metricRegistry, perPathMetricRegistry, authorizationService),
+                Ordered.LOWEST_PRECEDENCE - 50);
```

Comment:
Why are you invoking this filter with lower precedence than the lowest one in our constants? Could break integration with other filters in the future. Also why exactly 50?

Reply:
Would be nice to store all these values somewhere, and name them based on their impact.

Reply:
See: https://github.com/zalando/nakadi/pull/1346#discussion_r749359591